src/symbolic_memory.ml

```diff
@@ -38,27 +38,30 @@ module M = struct
     | _ -> Log.err {|Unsupported symbolic value reasoning over "%a"|} Expr.pp v
 
   let grow m delta =
-    let delta = Int32.to_int @@ i32 delta in
-    let old_size = m.size * page_size in
-    m.size <- max m.size ((old_size + delta) / page_size)
+    let old_size = Value.I32.mul m.size page_size in
+    let new_size = Value.I32.(div (add old_size delta) page_size) in
+    m.size <-
+      Triop (Ite, Value.I32.gt new_size old_size, new_size, old_size) @: Ty_bool
 
-  let size { size; _ } = Value.const_i32 @@ Int32.of_int (size * page_size)
+  let size { size; _ } = Value.I32.mul size page_size
 
-  let size_in_pages { size; _ } = Value.const_i32 @@ Int32.of_int @@ size
+  let size_in_pages { size; _ } = size
 
   let fill _ = assert false
 
   let blit _ = assert false
 
   let blit_string m str ~src ~dst ~len =
-    (* Always concrete? *)
+    (* This function is only used in memory init so everything will be concrete *)
+    let str_len = String.length str in
+    let mem_len = Int32.(to_int (i32 m.size) * to_int (i32 page_size)) in
     let src = Int32.to_int @@ i32 src in
     let dst = Int32.to_int @@ i32 dst in
     let len = Int32.to_int @@ i32 len in
     if
-      src < 0 || dst < 0
-      || src + len > String.length str
-      || dst + len > m.size * page_size
+      src < 0 || dst < 0 || len < 0
```

Comment:
Why is this needed ?

Reply:
It's checking for out of bounds. I tried being loyal to the [concrete version](https://github.com/OCamlPro/owi/blob/main/src/concrete_memory.ml#L58)

Reply:
Oh indeed. This should probably be moved into the interpreter anyway (I believe there's some `TODO` in the code).

Reply:
Yes there is. I can first make a PR moving the out of bounds checks of `Memory.blit_string`, `Memory.blit`, and `Memory.fill` to the interpreter if you'd like.